lib/sentry/opentelemetry/span_storage.ex

```diff
@@ -0,0 +1,151 @@
+defmodule Sentry.OpenTelemetry.SpanStorage do
+  @moduledoc false
+  use GenServer
+
+  defstruct [:cleanup_interval, :table_name]
+
+  alias Sentry.OpenTelemetry.SpanRecord
+
+  @cleanup_interval :timer.minutes(5)
+  @span_ttl :timer.minutes(30)
+
+  @spec start_link(keyword()) :: GenServer.on_start()
+  def start_link(opts) when is_list(opts) do
+    name = Keyword.get(opts, :name, __MODULE__)
+    GenServer.start_link(__MODULE__, opts, name: name)
+  end
+
+  @impl true
+  def init(opts) do
+    table_name = Keyword.get(opts, :table_name, default_table_name())
+    cleanup_interval = Keyword.get(opts, :cleanup_interval, @cleanup_interval)
+
+    _ = :ets.new(table_name, [:named_table, :public, :ordered_set])
+
+    schedule_cleanup(cleanup_interval)
+
+    {:ok, %__MODULE__{cleanup_interval: cleanup_interval, table_name: table_name}}
+  end
+
+  @impl true
+  def handle_info(:cleanup_stale_spans, state) do
+    cleanup_stale_spans(state.table_name)
+    schedule_cleanup(state.cleanup_interval)
+
+    {:noreply, state}
+  end
+
+  @spec store_span(SpanRecord.t(), keyword()) :: true
+  def store_span(span_data, opts \\ [])
+
+  def store_span(span_data, opts) when span_data.parent_span_id == nil do
+    table_name = Keyword.get(opts, :table_name, default_table_name())
+    stored_at = System.system_time(:second)
+
+    :ets.insert(table_name, {{:root_span, span_data.span_id}, span_data, stored_at})
+  end
+
+  def store_span(span_data, opts) do
+    table_name = Keyword.get(opts, :table_name, default_table_name())
+
+    stored_at = System.system_time(:second)
+    key = {:child_span, span_data.parent_span_id, span_data.span_id}
+
+    :ets.insert(table_name, {key, span_data, stored_at})
+  end
+
+  @spec get_root_span(String.t(), keyword()) :: SpanRecord.t() | nil
+  def get_root_span(span_id, opts \\ []) do
+    table_name = Keyword.get(opts, :table_name, default_table_name())
+
+    case :ets.lookup(table_name, {:root_span, span_id}) do
+      [{{:root_span, ^span_id}, span, _stored_at}] -> span
+      [] -> nil
+    end
+  end
+
+  @spec get_child_spans(String.t(), keyword()) :: [SpanRecord.t()]
+  def get_child_spans(parent_span_id, opts \\ []) do
+    table_name = Keyword.get(opts, :table_name, default_table_name())
+
+    :ets.match_object(table_name, {{:child_span, parent_span_id, :_}, :_, :_})
+    |> Enum.map(fn {_key, span_data, _stored_at} -> span_data end)
```

Comment:
You can change the match spec to only return `span_data` here and not have to do the `Enum.map/2`.